frontend/src/components/custom-tools/prompt-card/PromptCardItems.jsx

```diff
@@ -78,10 +80,18 @@ function PromptCardItems({
   const isNotSingleLlmProfile = llmProfiles.length > 1;
   const divRef = useRef(null);
   const [enforceType, setEnforceType] = useState("");
-  const profileId = singlePassExtractMode
-    ? defaultLlmProfile
-    : selectedLlmProfileId || defaultLlmProfile;
-  const coverageKey = generateCoverageKey(promptDetails?.prompt_id, profileId);
+  const promptId = promptDetails?.prompt_id;
+  const docId = selectedDoc?.document_id;
+  const promptProfile = promptDetails?.profile_manager || defaultLlmProfile;
+  const promptOutputKey = generatePromptOutputKey(
+    promptId,
+    docId,
+    promptProfile,
+    singlePassExtractMode,
+    true
+  );
+  const promptCoverage =
+    promptOutputs[promptOutputKey]?.coverage || coverageCountData;
```

Comment:
@jagadeeswaran-zipstack This is a slight corner case here. The value of coverage can be `0`, which is valid. However, if `promptOutputs[promptOutputKey]?.coverage` is `0`, it will be considered as false and fall back to the value of `coverageCountData`, which is incorrect.

To fix this, you can use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) (`??`) instead of `||`. Here's the code:

```suggestion
  const promptCoverage = promptOutputs[promptOutputKey]?.coverage ?? coverageCountData;
```

This ensures that the fallback value is only used if `promptOutputs[promptOutputKey]?.coverage` is `null` or `undefined`, and valid values like `0` are handled correctly.

Reply:
@tahierhussain `promptOutputs[promptOutputKey]?.coverage` returns array of docId's not actual count that's why we have this logic. Like i mentioned if there is no output for a prompt then it will fallback to initial store where we store coverage data.